client/src/components/JobList.tsx

```diff
@@ -20,15 +36,50 @@ export default function JobList() {
     loadJobs();
   }, []);
 
+  /**
+   * This block filters the jobs list based on the inputs from FilterForm.tsx
+   * There is a ternary for each input field.
+   * Salary ternary handles the number to string comparison;
+   *   The select options for salary increments by 50000, hence line 55
+   */
+  const filteredJobs = jobs?.filter(
+    (item) =>
+      (searchInput
```

Comment:
I would extract each of these checks to a separate function. So it then looks something like:
```
matchesSearch(searchInput, item) 
&& matchesLocation(location, item) 
&& matchesLevel(levelSelect, item) 
&& matchesSalary(salarySelect, item)
```

Reply:
Ahh okay, that reads so much better! This is will help a lot.